konnector.js

```diff
@@ -1,91 +1,229 @@
+const moment = require('moment')
+
+const {log, baseKonnector, filterExisting, saveDataAndFile, linkBankOperation, models} = require('cozy-konnector-libs')
+const Bill = models.bill
+
+const fileOptions = {
+  vendor: 'UBER',
+  dateFormat: 'YYYYMMDD'
+}
 
 module.exports = baseKonnector.createNew({
-  name: 'io.cozy.kitten',
-  models: [],
+  name: 'Uber',
+  vendorLink: 'https://uber.com',
+
+  dataType: ['bill'],
+  models: [Bill],
+
+  category: 'transport',
+  color: {
+    hex: '#000203',
+    css: '#000203'
+  },
+
   // fetchOperation is the list of function which will be called in sequence with the following
   // parameters :
   // requiredFields : the list of attributes of your connector that the user can choose (ofter login and password)
   // entries : it is an object which is passed accross the functions
   // data : another object passed accross function, not used
   // next : this is a callback you have to call when the task of the current function is finished
   fetchOperations: [
-    fetchKittens,
+    logIn,
+    getTrips,
     customFilterExisting,
-    customSaveDataAndFile
+    customSaveDataAndFile,
+    logOut,
+    linkBankOperation({
+      log,
+      minDateDelta: 1,
+      maxDateDelta: 1,
+      model: Bill,
+      amountDelta: 0.1,
+      identifier: 'UBER'
+    })
   ]
 })
 
-function fetchKittens (requiredFields, entries, data, next) {
-  let request = require('request')
-  // request-debug is a module which can trace and display in the standard output all the internet
-  // requests made by the konnector with the request module or request-promise also.
-  // require('request-debug')(request)
-
-  // Here we define the default parameters used in any http request made by the request modules so
-  // that you do not have to fill it every time. Here we only do one request but it can be usefull
-  // for a real connector.
-  request = request.defaults({
-    json: true,
-    headers: {
-      // a lot of web service do not want to be called by robots and then check the user agent to
-      // be sure they are called by a browser. This user agent works most of the time.
-      'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:36.0) ' +
-                    'Gecko/20100101 Firefox/36.0'
+function logIn (requiredFields, bills, data, next) {
+  const logInOptions = {
+    method: 'GET',
+    jar: true,
+    url: 'https://login.uber.com/login'
+  }
+
+  request(logInOptions, (err, res, body) => {
+    if (err) return next(err)
+
+    const $ = cheerio.load(body)
+    const token = $('input[name=_csrf_token]').val()
+
+    const signInOptions = {
+      method: 'POST',
+      jar: true,
+      url: 'https://login.uber.com/login',
+      form: {
+        email: requiredFields.login,
+        password: requiredFields.password,
+        _csrf_token: token
+      }
+    }
+
+    log.info('Logging in')
+
+    return request(signInOptions, (err, res) => {
+      if (err) {
+        log.error('Login failed')
```

Comment:
There is now a specific error code to say that the login has failed : `LOGIN_FAILED`. You can find the long list of codes [here](https://github.com/cozy/cozy-konnector-template/blob/master/docs/ERROR_CODES.md).

This code will be interpreted by cozy-collect to display a specific error message.